Validation/HGCalValidation/test/HGCMissingRecHit.cc

```diff
@@ -0,0 +1,458 @@
+                                        std::map<unsigned int, HGCHitTuple> const &hitRefs) {
+  std::vector<unsigned int> ids;
+  for (auto it = theHits->begin(); it != theHits->end(); ++it)
+    ids.emplace_back((it->id().rawId()));
+  for (auto it = hitRefs.begin(); it != hitRefs.end(); ++it) {
+    double eta = getEta(it->second);
+    auto itr = std::find(ids.begin(), ids.end(), it->first);
+    if (itr == ids.end()) {
+      missedHitsDE_[idet]->Fill(std::get<0>(it->second));
+      missedHitsDT_[idet]->Fill(eta);
+      std::ostringstream st1;
+      if (DetId(it->first).det() == DetId::HGCalHSc)
+        st1 << HGCScintillatorDetId(it->first);
+      else
+        st1 << HGCSiliconDetId(it->first);
+      edm::LogVerbatim("HGCalMiss") << "Hit: " << std::hex << (it->first) << std::dec << " " << st1.str()
+                                    << " SimHit (E = " << std::get<0>(it->second) << ", X = " << std::get<1>(it->second)
+                                    << ", Y = " << std::get<2>(it->second) << ", Z = " << std::get<3>(it->second)
+                                    << ") is missing in the Digi collection";
+    } else {
+      goodHitsDE_[idet]->Fill(std::get<0>(it->second));
+      goodHitsDT_[idet]->Fill(eta);
+    }
+  }
+}
+
+template <class T1>
+void HGCMissingRecHit::analyzeHGCalRecHit(T1 const &theHits,
+                                          int idet,
+                                          std::map<unsigned int, HGCHitTuple> const &hitRefs) {
+  std::vector<unsigned int> ids;
+  for (auto it = theHits->begin(); it != theHits->end(); ++it)
+    ids.emplace_back((it->id().rawId()));
+  for (auto it = hitRefs.begin(); it != hitRefs.end(); ++it) {
+    double eta = getEta(it->second);
+    auto itr = std::find(ids.begin(), ids.end(), it->first);
+    if (itr == ids.end()) {
+      missedHitsRE_[idet]->Fill(std::get<0>(it->second));
+      missedHitsRT_[idet]->Fill(eta);
+      std::ostringstream st1;
+      if (DetId(it->first).det() == DetId::HGCalHSc)
+        st1 << HGCScintillatorDetId(it->first);
+      else
+        st1 << HGCSiliconDetId(it->first);
+      edm::LogVerbatim("HGCalMiss") << "Hit: " << std::hex << (it->first) << std::dec << " " << st1.str()
+                                    << " SimHit (E = " << std::get<0>(it->second) << ", X = " << std::get<1>(it->second)
+                                    << ", Y = " << std::get<2>(it->second) << ", Z = " << std::get<3>(it->second)
+                                    << ") is missing in the RecHit collection";
+    } else {
+      goodHitsRE_[idet]->Fill(std::get<0>(it->second));
+      goodHitsRT_[idet]->Fill(eta);
+    }
+  }
+}
+
+double HGCMissingRecHit::getEta(const HGCHitTuple &hitRef) {
+  double x = std::get<1>(hitRef);
+  double y = std::get<2>(hitRef);
+  double r = std::sqrt(x * x + y * y);
+  double z = std::abs(std::get<3>(hitRef));
+  double theta = std::atan(r / z);
+  double eta = (z > 0) ? -std::log(std::tan(0.5 * theta)) : 100.0;
```

Comment:
Is there any particular reason to assign a value of 100 as `eta` to all points with negative z coordinate?